quinn-proto/src/frame.rs

```diff
@@ -928,6 +948,118 @@ impl AckFrequency {
     }
 }
 
+/* Address Discovery https://datatracker.ietf.org/doc/draft-seemann-quic-address-discovery/ */
+
+/// Conjuction of the information contained in the address discovery frames ([`ObservedIpV4Addr`],
+/// [`ObservedIpV6Addr`]).
+#[derive(Debug)]
+pub(crate) struct ObservedAddr {
+    ip: IpAddr,
+    port: u16,
+    request_id: u64,
+}
+
+/// Frame for an observed ipv4 address.
+///
+/// This corresponds to [`Type::OBSERVED_IPV4_ADDR`].
+pub(crate) struct ObservedIpV4Addr {
+    /// Random request id for debugging.
+    ///
+    /// NOTE(@divma): this is an assumption I make since this is not defined anywhere and STUN
+    /// indications use a random id as well.
```

Comment:
IIRC in STUN this is used to match requests to responses.  Here however there is never a request as these frames are eagerly pushed by the observer.  Does this still help adding this here?  When have you used it?

Reply:
STUN uses request and responses _and indications_. They all have request ids, and indications are not responses to any request. This is my guess to the intention since the draft is.. very much a draft, missing definitions and references. Part of what's missing is the purpose of this field. I'm keeping it for now because it's there and it's arguably similar to something it intends to emulate. I raised this question on the repo for the draft. At the end, if I do find it useless it can be removed